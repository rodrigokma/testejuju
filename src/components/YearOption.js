function YearOption(props) {
  return (
    <div className="year-option">
      {props.modelCode && (
        <select value={props.value} onChange={props.onChange}>
          <option data-testid="select-year" value="">
            Choose a year
          </option>

          {props.years.map((year) => {
            return (
              <option key={`year-${year.codigo}`} value={year.codigo}>
                {year.nome}
              </option>
            );
          })}
        </select>
      )}
    </div>
  );
}

export default YearOption;
