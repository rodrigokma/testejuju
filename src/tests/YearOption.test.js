import React from "react";
import YearOption from "../components/YearOption";
import { render } from "@testing-library/react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import TestRenderer from "react-test-renderer";

Enzyme.configure({ adapter: new Adapter() });

describe("Year options component tests", () => {
  const yearOptionJsx = (
    <YearOption
      modelCode="5276"
      onChange={jest.fn()}
      years={[
        { nome: "2013 gasolina", codigo: "2013-1" },
        { nome: "2012 gasolina", codigo: "2012-1" },
        { nome: "2011 gasolina", codigo: "2011-1" },
      ]}
    />
  );
  const component = shallow(yearOptionJsx);

  test("renders component", () => {
    expect(component.exists()).toBe(true);
  });

  test("matches the snapshot", () => {
    const componentSnapshot = TestRenderer.create(yearOptionJsx);
    expect(componentSnapshot).toMatchSnapshot();
  });

  test("has a 'Choose a year' as first option", () => {
    const { getByTestId } = render(yearOptionJsx);

    expect(getByTestId("select-year")).toHaveTextContent("Choose a year");
  });

  test("Renders three year options", () => {
    expect(component.find("option")).toHaveLength(4);
  });
});
