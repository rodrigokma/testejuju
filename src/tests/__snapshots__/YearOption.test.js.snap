// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Year options component tests matches the snapshot 1`] = `
<div
  className="year-option"
>
  <select
    onChange={[MockFunction]}
  >
    <option
      data-testid="select-year"
      value=""
    >
      Choose a year
    </option>
    <option
      value="2013-1"
    >
      2013 gasolina
    </option>
    <option
      value="2012-1"
    >
      2012 gasolina
    </option>
    <option
      value="2011-1"
    >
      2011 gasolina
    </option>
  </select>
</div>
`;
